Update Ubuntu install instructions for kilted

Crola1702 opened the pull request:
This PR changes the instructions from manually setting the repositories to use [ros-apt-source](https://github.com/ros-infrastructure/ros-apt-source/) package instead.

Crola1702 committed:
Update install instructions for kilted

Signed-off-by: Crola1702 <cristobal.arroyo@ekumenlabs.com>

Crola1702 committed:
Fix title underline

Signed-off-by: Crola1702 <cristobal.arroyo@ekumenlabs.com>

nuclearsandwich commented on source/Installation/Ubuntu-Install-Debs.rst:
```diff
@@ -32,8 +32,8 @@ Set locale
 
-Enable required repositories
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Install ros-apt-source package
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
I think it's okay to leave the title as it was. We're still enabling the required repositories, only the how is changing.

nuclearsandwich commented on source/Installation/_Apt-Repositories.rst:
```diff
@@ -1,21 +1,8 @@
-You will need to add the ROS 2 apt repository to your system.
+The package `ros-apt-source <https://github.com/ros-infrastructure/ros-apt-source/>`_ provides keys and source configuration for the ROS repositories.
```
Again, let's leave the explanatory guidance intact and add to it rather than replace it.

nuclearsandwich commented on source/Installation/_Apt-Repositories.rst:
```diff
@@ -1,21 +1,8 @@
-You will need to add the ROS 2 apt repository to your system.
+The package `ros-apt-source <https://github.com/ros-infrastructure/ros-apt-source/>`_ provides keys and source configuration for the ROS repositories.
 
-First ensure that the `Ubuntu Universe repository <https://help.ubuntu.com/community/Repositories/Ubuntu>`_ is enabled.
```
This requirement still stands so this line should be preserved.

nuclearsandwich commented on source/Installation/_Apt-Repositories.rst:
```diff
@@ -1,21 +1,8 @@
+The package `ros-apt-source <https://github.com/ros-infrastructure/ros-apt-source/>`_ provides keys and source configuration for the ROS repositories.
 
-First ensure that the `Ubuntu Universe repository <https://help.ubuntu.com/community/Repositories/Ubuntu>`_ is enabled.
+You will need install this package con configure ROS repositories.
```
```suggestion
Installing the ros2-apt-source package will configure ROS 2 repositories for your system.
Updates to repository configuration will occur automatically when new versions of this package are released to the ROS repositories.
```

Let's add a note explaining what's happening and also mention that updates to repository configuration are now smoother.

christophebedard replied:
```suggestion
You will need to install this package to configure ROS apt repositories.
```

Crola1702 committed:
Update source/Installation/_Apt-Repositories.rst

Co-authored-by: Christophe Bedard <bedard.christophe@gmail.com>
Signed-off-by: Cristóbal Arroyo <j.arroyo@uniandes.edu.co>

christophebedard commented on source/Installation/_Apt-Repositories.rst:
```diff
@@ -1,21 +1,8 @@
-.. code-block:: console
-
-   $ echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2-testing/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | sudo tee /etc/apt/sources.list.d/ros2.list > /dev/null
+   $ curl -O --output-dir /tmp/ https://ftp.osuosl.org/pub/ros/packages.ros.org/ros2-testing/ubuntu/pool/main/r/ros-apt-source/ros2-apt-source_1.0.0~$(. /etc/os-release && echo $VERSION_CODENAME)_all.deb
```
We switched to the testing repo for the Kilted pre-release validation (e.g., tutorial party). Would the corresponding package be `ros2-testing-apt-source_*` instead here?

christophebedard replied:
I see that https://ftp.osuosl.org/pub/ros/packages.ros.org/ros2-testing/ubuntu/pool/main/r/ros-apt-source/ (which includes "testing" in the URL) contains both testing and non-testing packages, but https://ftp.osuosl.org/pub/ros/packages.ros.org/ros2/ubuntu/pool/main/r/ros-apt-source/ (no "testing" in the URL) also exists and contains the same packages.

christophebedard replied:
oh, nevermind, Steven! is already reviewing this

Crola1702 replied:
Yes, you're right

nuclearsandwich commented on source/Installation/_Apt-Repositories.rst:
```diff
@@ -1,21 +1,8 @@
-.. code-block:: console
-
-   $ echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2-testing/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | sudo tee /etc/apt/sources.list.d/ros2.list > /dev/null
+   $ curl -O --output-dir /tmp/ https://ftp.osuosl.org/pub/ros/packages.ros.org/ros2-testing/ubuntu/pool/main/r/ros-apt-source/ros2-apt-source_1.0.0~$(. /etc/os-release && echo $VERSION_CODENAME)_all.deb
+   $ sudo apt install /tmp/ros2-apt-source_1.0.0~$(. /etc/os-release && echo $VERSION_CODENAME)_all.deb
```
```suggestion
   $ curl -o /tmp/ros2-testing-apt-source.deb https://ftp.osuosl.org/pub/ros/packages.ros.org/ros2-testing/ubuntu/pool/main/r/ros-apt-source/ros2-testing-apt-source_1.0.0~$(. /etc/os-release && echo $VERSION_CODENAME)_all.deb
   $ sudo apt install /tmp/ros2-testing-apt-source.deb
```

Since we plan to simplify the url with symlinks to the actual repo files (and mask the version number to reduce update toil) let's download using a generic name.

Also, use the testing repo package instead of the ros2-apt-source main package for the tutorial party.

Crola1702 committed:
Apply PR comments

Signed-off-by: Crola1702 <cristobal.arroyo@ekumenlabs.com>

Crola1702 committed:
Restore apt 2.4.13 (amd64)
Usage: apt [options] command

apt is a commandline package manager and provides commands for
searching and managing as well as querying information about packages.
It provides the same functionality as the specialized APT tools,
like apt-get and apt-cache, but enables options more suitable for
interactive use by default.

Most used commands:
  list - list packages based on package names
  search - search in package descriptions
  show - show package details
  install - install packages
  reinstall - reinstall packages
  remove - remove packages
  autoremove - Remove automatically all unused packages
  update - update list of available packages
  upgrade - upgrade the system by installing/upgrading packages
  full-upgrade - upgrade the system by removing/installing/upgrading packages
  edit-sources - edit the source information file
  satisfy - satisfy dependency strings

See apt(8) for more information about the available commands.
Configuration options and syntax is detailed in apt.conf(5).
Information about how to configure sources can be found in sources.list(5).
Package and version choices can be expressed via apt_preferences(5).
Security details are available in apt-secure(8).
                                        This APT has Super Cow Powers. word

Signed-off-by: Crola1702 <cristobal.arroyo@ekumenlabs.com>

christophebedard commented on source/Installation/_Apt-Repositories.rst:
```diff
@@ -7,15 +7,12 @@ First ensure that the `Ubuntu Universe repository <https://help.ubuntu.com/commu
-
-   $ sudo apt update && sudo apt install curl -y
-   $ sudo curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg
-
-Then add the repository to your sources list.
+Installing the ros2-apt-source package will configure ROS 2 repositories for your system.
+Updates to repository configuration will occur automatically when new versions of this package are released to the ROS repositories.
```
Very nitpicky, feel free to ignore: we use both "ros-apt-source" and "ros2-apt-source" here. Would one of them be more correct?

nuclearsandwich replied:
We haven't sent documentation updates describing the different packages that exist.

* ros2-apt-source configures the main ROS 2 repository
* ros2-testing-apt-source configures the ROS 2 testing repository
* ros-apt-source (on Ubuntu Focal only) configures the ROS main repository
* ros-testing-apt-source (on Ubuntu Focal only) configures the ROS testing repository

christophebedard replied:
I think the source of my nitpicking is that we refer to the whole set of packages as "the ros-apt-source package" because that's the repo name (https://github.com/ros-infrastructure/ros-apt-source/).

I apparently can't do a suggestion here, but what about:

```diff
-The package `ros-apt-source <https://github.com/ros-infrastructure/ros-apt-source/>`_ provides keys and source configuration for the ROS repositories.
+The `ros-apt-source <https://github.com/ros-infrastructure/ros-apt-source/>`_ packages provide keys and source configuration for the ROS repositories.
```

christophebedard replied:
Anyway, we can proceed with this if it looks good to you @nuclearsandwich

nuclearsandwich replied:
@christophebedard ah thanks for clarifying! I agree and made a similar suggestion inline above.

nuclearsandwich commented on source/Installation/_Apt-Repositories.rst:
```diff
@@ -7,15 +7,12 @@ First ensure that the `Ubuntu Universe repository <https://help.ubuntu.com/commu
    $ sudo add-apt-repository universe
 
-Now add the ROS 2 GPG key with apt.
+The package `ros-apt-source <https://github.com/ros-infrastructure/ros-apt-source/>`_ provides keys and source configuration for the ROS repositories.
```
```suggestion
The `ros-apt-source <https://github.com/ros-infrastructure/ros-apt-source/>`_ packages provide keys and apt source configuration for the various ROS repositories.
```

nuclearsandwich reviewed (approved):
Good to go once @christophebedard's point about distinguishing between the collection of "ros-apt-source" packages and the individual `ros{,2}{,-testing}-apt-source` package being installed. Either of our suggestions is sufficient to resolve this comment.

Crola1702 committed:
Update source/Installation/_Apt-Repositories.rst

Co-authored-by: Steven! Ragnarök <nuclearsandwich@users.noreply.github.com>
Signed-off-by: Cristóbal Arroyo <j.arroyo@uniandes.edu.co>
